#! /bin/sh -e

# All lines beginning with `# DP:' are a description of the patch.
# DP: Description: Remove stray .prologue
# DP: Author: Phil Blundell <pb@debian.org>

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p0 < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p0 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# append the patch here and adjust the -p? flag in the patch calls.

--- linuxthreads/sysdeps/alpha/elf/pt-initfini.c	5 Jul 2003 22:56:39 -0000	1.3
+++ linuxthreads/sysdeps/alpha/elf/pt-initfini.c	29 Jul 2003 19:15:35 -0000
@@ -75,7 +75,6 @@
 	subq	$30, 16, $30				\n\
 	stq	$26, 0($30)				\n\
 	stq	$29, 8($30)				\n\
-	.prologue 1					\n\
 	.align 3					\n\
 /*@_fini_PROLOG_ENDS*/					\n\
 							\n\
